DuggaSys/diagram.js

```diff
@@ -476,7 +473,12 @@ class StateMachine
         // Change the sliders max to historyLogs length
         document.getElementById("replay-range").setAttribute("max", this.historyLog.length.toString());
     }
-
+    removeFutureStates(){
+        // Remove the history entries that are after current index
+        while(this.currentHistoryIndex + 1 != this.historyLog.length) {
+            this.historyLog.pop();
+        }
```

Comment:
Wouldn't just a splice do? Same end result but doesn't need to resize the object more than once that way.

Reply:
It is true, we just used the old code that did the removal of elements before currentIndex, do you want us to change to an splice?

Reply:
Preferably. Less code with same result.

Reply:
The changes are done. Can you solve the conflict and then request an review again from @a19steis